### PR TITLE
Fixes part of #4195:Added dark mode support to QuestionPlayer and ExplorationPlayer

### PR DESCRIPTION
**Pull request by bhaktideshmukh**
## Explanation 

This PR contains the changes of PR [#4382](https://github.com/oppia/oppia-android/pull/4382) and the pending cases of this issue such as drag_drop_interaction, forward_arrow etc.

**Mock link** :--

-  https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/249c3349-605b-4569-bbe9-1a017ab90101/specs/
- https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/f2d5e28f-204d-49ef-ab16-430ec335e35e/specs/

**Screenshots**

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by bhaktideshmukh** [comment]
Here in this PR, I have not changed the cursor color as it was in PR https://github.com/oppia/oppia-android/pull/4382 since I figured out that the changes made in `theme.xml` in order to change the cursor color lead to the app crashing.
Maybe we can create a separate issue handling this issue.
What are your thoughts @rt4914  @BenHenning ?

**Comment by BenHenning** [comment]
> Here in this PR, I have not changed the cursor color as it was in PR #4382 since I figured out that the changes made in `theme.xml` in order to change the cursor color lead to the app crashing. Maybe we can create a separate issue handling this issue. What are your thoughts @rt4914 @BenHenning ? 

@bhaktideshmukh I'm not opposed to a separate issue tracking the specific problem so long as what we check in here doesn't break the UI/UX or crash the app. That being said, what exactly is the crash? Is there a debug doc that sheds light on it? Ideally we would keep all related dark mode changes together, but this might be a good case of separating this specific change if it's particularly challenging.

File: app/src/main/res/values/color_defs.xml [hunk 1]
```diff
@@ -71,4 +71,9 @@
   <color name="color_def_dark_mode_question_player_progress_bar_gradient_end">#00C7B6</color>
   <color name="color_def_whitish_grey">#E5E5E5</color>
   <color name="color_def_checkbox_green">#008577</color>
+  <color name="color_def_oppia_dark_grey_border">#666666</color>
```
Comment (rt4914) [inline app/src/main/res/values/color_defs.xml]: This color is already defined as `color_def_accessible_light_grey_2 `. You can directly use that.

File: app/src/main/res/values/color_defs.xml [hunk 2]
```diff
@@ -71,4 +71,9 @@
...
+  <color name="color_def_stroke_silver">#777777</color>
+  <color name="color_def_drag_drop_single_item_gradient_start">#FFFFFF</color>
```
Comment (rt4914) [inline app/src/main/res/values/color_defs.xml]: This color is also defined as `color_def_white `

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh Suggested changes.

**Comment by bhaktideshmukh** [comment]
> > Here in this PR, I have not changed the cursor color as it was in PR #4382 since I figured out that the changes made in `theme.xml` in order to change the cursor color lead to the app crashing. Maybe we can create a separate issue handling this issue. What are your thoughts @rt4914 @BenHenning ?
> 
> @bhaktideshmukh I'm not opposed to a separate issue tracking the specific problem so long as what we check in here doesn't break the UI/UX or crash the app. That being said, what exactly is the crash? Is there a debug doc that sheds light on it? Ideally we would keep all related dark mode changes together, but this might be a good case of separating this specific change if it's particularly challenging.

@BenHenning What I figured out while debugging was when I added
 `<item name="android:textCursorDrawable">@style/InputInteractionEditText</item>` this in `theme.xml` resulted in app crashing as mentioned in issue [#4477](https://github.com/oppia/oppia-android/issues/4477). 
I even checked logs and found an error stating 'Listener not found' that's the reason I was suggesting creating a new separate issue for this. For now, there is no such debugging doc describing this issue.

File: app/src/main/res/values/color_defs.xml (hunk 1 above)
Comment (bhaktideshmukh) [reply app/src/main/res/values/color_defs.xml]: > This color is already defined as `color_def_accessible_light_grey_2 `. You can directly use that.

Yes, correct.

File: app/src/main/res/values/color_defs.xml (hunk 2 above)
Comment (bhaktideshmukh) [reply app/src/main/res/values/color_defs.xml]: > This color is also defined as `color_def_white `

Done.

**Comment by BenHenning** [comment]
@bhaktideshmukh did you figure out the root cause of the crash? Without understanding why the change causes the crash it's difficult to make decisions about alternative solutions.

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh did you figure out the root cause of the crash? Without understanding why the change causes the crash it's difficult to make decisions about alternative solutions.

Yes @BenHenning you are correct without knowing the actual root cause we shouldn't proceed with the alternatives but I was unable to figure out why adding something externally in theme.xml lead to crash.
In PR #4382, I followed [this](https://stackoverflow.com/questions/7238450/set-edittext-cursor-color/7238526#7238526) in order to change the cursor color.

**Comment by BenHenning** [comment]
@bhaktideshmukh thanks for the details. Per the SO answer, the android:textCursorDrawable is supposed to be set to a drawable, but I think you set it to a style. Only android:editTextStyle was set to a style in the answer. Is it possible the solution was implemented incorrectrly?

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh thanks for the details. Per the SO answer, the android:textCursorDrawable is supposed to be set to a drawable, but I think you set it to a style. Only android:editTextStyle was set to a style in the answer. Is it possible the solution was implemented incorrectrly?

@BenHenning Yes you were correct, the solution was incorrectly implemented and I apologize for that.
Now I have successfully implemented it.

| Default | Dark mode |
|-------|---|
|||

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**